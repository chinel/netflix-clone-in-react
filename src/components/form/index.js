import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
