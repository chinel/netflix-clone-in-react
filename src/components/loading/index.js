import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
