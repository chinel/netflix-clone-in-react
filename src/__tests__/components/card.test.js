import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Card, Player } from "../../components";

const category = "series";
const slideRows = [
  {
    title: "Documentaries",
    data: [
      {
        genre: "documentaries",
        maturity: "18",
        slug: "tiger-king",
        description: "Tiger King description",
        id: "07adcf19-517f-41b4-8f8b-04fee694bce1",
        title: "Tiger King",
        docId: "IcRxJBanzE7nmIp2GMGb",
      },
    ],
  },
  {
    title: "Feel Good",
    data: [
      {
        title: "Juno",
        genre: "feel-good",
        description: "Juno description",
        maturity: "0",
        id: "a938b5a6-fe25-4a06-8050-353bc7146ccd",
        slug: "juno",
        docId: "4JDgdf5vE0K3YrW9ZnbP",
      },
    ],
  },
];

describe("<Card/>", () => {
  it("renders the <Card /> with populated data", () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );
  });
});
