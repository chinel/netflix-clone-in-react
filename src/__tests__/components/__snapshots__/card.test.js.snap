// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders the <Card /> with populated data 1`] = `
<div
  class="card__Group-sc-18hcxw-2 cCJyil"
>
  <div
    class="card__Container-sc-18hcxw-1 bIBsBm"
  >
    <p
      class="card__Title-sc-18hcxw-0 bgqEna"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-18hcxw-12 fYIIpv"
    >
      <div
        class="card__Item-sc-18hcxw-13 gbOoym"
      >
        <img
          class="card__Image-sc-18hcxw-14 izLMEn"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-18hcxw-11 gRwPQP"
        >
          <p
            class="card__SubTitle-sc-18hcxw-3 jfWBKr"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-18hcxw-4 dmUYan"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-18hcxw-1 bIBsBm"
  >
    <p
      class="card__Title-sc-18hcxw-0 bgqEna"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-18hcxw-12 fYIIpv"
    >
      <div
        class="card__Item-sc-18hcxw-13 gbOoym"
      >
        <img
          class="card__Image-sc-18hcxw-14 izLMEn"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-18hcxw-11 gRwPQP"
        >
          <p
            class="card__SubTitle-sc-18hcxw-3 jfWBKr"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-18hcxw-4 dmUYan"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
